### PR TITLE
keys: Don't panic when error can be returned

### LINKED ISSUE #110752: keys: crash in PrettyPrint and DecodeTenantPrefix on invalid keys
**Describe the problem**

Found by @miretskiy using fuzzing: if the input key does't contain a valid integer in the tenant prefix (e.g. the key was generated by PrefixEnd), these functions crash.

**Expected behavior**

regular error instead.

Epic: CRDB-26091

Jira issue: CRDB-31590

### PR DESCRIPTION
**Pull request by miretskiy**
Avoid panic when trying to decode key.  We have
to assume that sometimes we may want to log
keys that may be corrupt.  Panic in this case
is unnecessary since the key pretty printer can
return an error instead.

Add a pretty printer fuzz test that discovered
this issue.  Validated over 1hr run that
pretty printer no longer panics/generates errors.

Epic: CRDB-26091
Fixes #110752.

Release note: None

### CODE REVIEW

**Review by knz — APPROVED** [review]
thank you!

File: pkg/keys/sql.go [hunk 1]
```diff
@@ -52,7 +52,11 @@ func DecodeTenantPrefix(key roachpb.Key) ([]byte, roachpb.TenantID, error) {
 	if err != nil {
 		return nil, roachpb.TenantID{}, err
 	}
-	return rem, roachpb.MustMakeTenantID(tenID), nil
+	id, err := roachpb.MakeTenantID(tenID)
+	if err != nil {
+		return nil, roachpb.SystemTenantID, err
```
Comment (knz) [inline pkg/keys/sql.go:57]: can you return `roachpb.TenantID{}` here thanks.

Comment (miretskiy) [reply pkg/keys/sql.go:57]: Sure.

**Review by knz — APPROVED** [review]

**Pull request merged**